reddit_gold/public/static/js/snoovatar/snoovatar.js

```diff
@@ -57,56 +62,42 @@
 
   /**
    * returns a promise that resolves when an array of image sources are preloaded
-   * @param  {string[]} srcs list of image sources to preload
+   * @param  {string[]} sources list of svg bundles to load}
    * @return {$.Promise}
-   * @resolve {Image[]}
+   * @resolve {JSON with SVG}
    */
-  $.preloadImageArray = function(srcs) {
-    return $.when.apply(window, $.map(srcs, function(src) {
-      return $.preloadImage(src);
-    }));
-  };
+  $.preloadSVGs = function(sources) {
+    return $.when.apply(window, $.map(sources, function(src) {
+      var isJson = _.isObject(src);
+      var def = $.Deferred();
+      r.ajax({
+        url: src.url,
+        type: 'GET'
+      })
+        .done(function(data, textStatus, jqXHR) {
+          var data = jqXHR.responseText;
+          if (isJson) {
+            try {
+              data = window.JSON.parse(data);
+            } catch (e) {}
+          }
```

Comment:
in case you wanna request individual SVG not all the browsers properly parse svg xml dialect, so it's safer to use **raw** response.